### PR TITLE
10.2 server

### PR DESCRIPTION
**Pull request by 9EOR9**
Travis and Appveyor test fixes

### CODE REVIEW

File: libmariadb/mariadb_lib.c [hunk 1]
```diff
@@ -2653,26 +2657,7 @@ mysql_optionsv(MYSQL *mysql,enum mysql_option option, ...)
...
-        _splitpath(module_filename,NULL, NULL, appname, NULL);
-      }
-#else
-      const char * appname = "";
-#endif
-      OPT_SET_VALUE_STR(&mysql->options, my_cnf_group, appname);
-      break;
-    }
-    OPT_SET_VALUE_STR(&mysql->options, my_cnf_group, (char *)arg1);
+    OPT_SET_VALUE_STR(&mysql->options, my_cnf_group, arg1 ? (char *)arg1 : "");
```
Comment (vuvova) [inline libmariadb/mariadb_lib.c]: Why? I think it was an important feature, contributing to the standard expected behavior of all MariaDB clients

File: libmariadb/mariadb_lib.c [hunk 2]
```diff
@@ -2537,7 +2537,11 @@ mysql_stat(MYSQL *mysql)
 int STDCALL
 mysql_ping(MYSQL *mysql)
 {
-  return ma_simple_command(mysql, COM_PING,0,0,0,0);
+  int rc;
+  rc= ma_simple_command(mysql, COM_PING, 0, 0, 0, 0);
+  if (rc && mysql->options.reconnect)
+    rc= ma_simple_command(mysql, COM_PING, 0, 0, 0, 0);
+  return rc;
 }
```
Comment (vuvova) [inline libmariadb/mariadb_lib.c]: Hmm, perhaps you can reconnect only on certain errors? Like, no need to reconnect if the first ping got "access denied"

File: libmariadb/mariadb_lib.c (hunk 1 above)
Comment (9EOR9) [reply libmariadb/mariadb_lib.c]: - program name detection failed on several platforms
- In libmysql the default configuration groups will be read if you specify NULL or ""(empty string), but not the programname group.

File: libmariadb/mariadb_lib.c (hunk 2 above)
Comment (9EOR9) [reply libmariadb/mariadb_lib.c]: Ok, maybe only test if error code was CR_SERVER_LOST ?! (like it was before)

Comment (vuvova) [reply libmariadb/mariadb_lib.c]: @9E0R9 yes. Why did you remove that (your commit had no explanation for removal)?

@vaintroub yes, when using one-time passwords (google authenticator, rsa securid, etc) one cannot reconnect using old credentials.

Comment (9EOR9) [reply libmariadb/mariadb_lib.c]: I think we should check these error codes:

CR_SERVER_GONE_ERROR (2006)
CR_SERVER_LOST (2013)

File: libmariadb/mariadb_lib.c (hunk 1 above)
Comment (vuvova) [reply libmariadb/mariadb_lib.c]: Well, this is exactly the point. the "standard expected behavior" is to read the default groups and the programname group. Now there's no way to do it. If a client specifies "" or NULL, C/C won't read the programname group. If the client specifies "programname" explicitly, C/C won't read default groups. So no way to get the standard behavior.

On what platforms did the program name detection fail?

Comment (9EOR9) [reply libmariadb/mariadb_lib.c]: The current behavior of Connector/C and libmysql is that configuration files will be read if either config_file or config_group was specified:
```
/* use default options */
  if (mysql->options.my_cnf_file || mysql->options.my_cnf_group)
```
So the initial behavior for mysql_options(mysql, MYSQL_OPT_READ_DEFAULT_GROUP, "") was reading default configuration files with default groups, but not the program name.

Please note that also the my.cnf parser is very basic and will fail, e.g. if your application name has special chars, e.g. "]client"

Comment (9EOR9) [reply libmariadb/mariadb_lib.c]: If I remember correctly, program name wasn't detected on Solaris (this was once fixed in master branch), but also when crosscompiling for android and arm.

**Pull request merged**